buildtest/config_schema.json

```diff
@@ -0,0 +1,56 @@
+{
+    "$id": "https://raw.githubusercontent.com/HPC-buildtest/buildtest-framework/devel/buildtest/config_schema.json",
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "Buildtest Configuration Schema Definition",
+    "type": "object",
+    "defs": {
+        "alphanumeric": {
+            "type": "string",
+            "pattern": "([a-zA-Z0-9_]|-)+"
+        },
+        "vars": {
+            "type": "object",
+            "minItems": 1,
+            "items": {
+                "type": "object",
+                "minItems": 1,            
+                "propertyNames": {
+                    "pattern": "^[A-Za-z_][A-Za-z0-9_]*$"
+                }
+            }        
+        }
+    },
+    "properties": {
+        "editor": { 
+            "type": "string",
+            "enum": ["vi","vim","nano","emacs"]
+        },
+        "queues": {
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "name": {"$ref": "#/defs/alphanumeric"},
+                    "description": {"type": "string"},
+                    "scheduler": {
+                        "type": "string",
+                        "enum": ["local", "slurm"]
+                    },
+                    "launcher": {
+                        "type": "string",
+                        "enum": ["local", "mpirun", "mpiexec", "sbatch", "srun"]
```

Comment:
This launcher variable I think would be specific to an executor (and not represented in a global config validation per say, but validation for that particular executor).